### ISSUE TITLE
Postman 4.5.1 - Behind proxy

### ISSUE THREAD

**Issue by DerFlash** [a85/POSTMan-Chrome-Extension#2225]
1. Postman Version: 4.5.0
2. App (Chrome app or Mac app): Mac app
3. OS details: macOS 10.11.5
4. Is the Interceptor on and enabled in the app: no
5. Did you encounter this recently, or has this bug always been there: recently (update)
6. Expected behavior: Should work behind proxy out of the box
7. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app):
8. Screenshots (if applicable)
9. Steps to reproduce the problem:

Put your mac behind a http proxy. Sending requests now for internal servers works just fine. When trying to reach external servers which need to be routed over the OS' http proxy, they will time out eventually with the error:

> Could not get any response
> There was an error connecting to XXX.
> Why this might have happened:
> The server couldn't send a response:
> Ensure that the backend is working properly
> Self-signed SSL certificates are being blocked:
> Fix this by turning off 'SSL certificate verification' in Settings > General
> Request timeout:
> Change request timeout in Settings > General

Using Postman 4.4.3 on the same mac works like a charm!

Btw: I could manage to get v4.5.0 working by running it from Terminal and setting a http_proxy before:

```
$ export http_proxy="http://XXX@XXX:XXX/"
$ /Applications/Postman.app/Contents/MacOS/Postman
```

So it seems that v4.5.0 is just not reading the system proxy correctly?!

**Comment by shamasis** [comment]
Can you try by settings your proxy configuration in environment variables? We are working on providing a configuration for this. The global proxy is inside chrome and we need to have ability to set or do separate configuration in app.

**Comment by DerFlash** [comment]
Hey shamasis. I'm not sure what you mean with setting it in environment variables. I did the export in the Terminal.app before running Postman from there, as I showed in the initial post.
I think this may be exactly what you describe as environment variable?!

I also was able to add the export directly to my '.bash_profile', so it would get set automatically/globally when I log into my mac.

But as I said, you did in fact already read the system proxy settings just fine in Postman 4.4.3 - without the need of chrome or any extra configuration in the native Postman app.
For sure a configuration to either use the system proxy OR also add a different proxy manually would be a plus. ;-)

As a osx/iOS developer myself, it may help if I reference the default method to grab the system proxy from within a native app. Maybe that helps:
https://developer.apple.com/library/mac/documentation/Networking/Reference/SCDynamicStoreCopySpecific/index.html#//apple_ref/doc/c_ref/SCDynamicStoreCopyProxies

Kind regards
Björn

**Comment by DavidGloverBBC** [comment]
I am also experiencing this.

**Comment by shamasis** [comment]
Added proxy support in Roadmap.
- having separate proxy configuration should be better than simply using system proxy
- should try and detect system proxy (but detecting browser / chrome proxy settings in native app will not be possible)

Will come back with ETA ASAP

**Comment by DerFlash** [comment]
Hey shamasis,

thanks for your quick response!

The first point should definitely be a feature, yes. Would be cool to be able to configure it optionally separately. But since this is a "nice to have", it may take some time for sure.

But I hope the later point will be handled as a more urgent position since it's in my eyes clearly a bug because it was working fine in the last version and is broken now.
Especially because you can't configure it yet, it should always try and detect the system proxy automatically.

Kind regards
Björn

**Comment by shamasis** [comment]
Yes. We will detect system settings and unblock first. Should be out in next release.

**Comment by DerFlash** [comment]
Perfect!

**Comment by shamasis** [comment]
We've addressed the issue. You would now be able to transparently use system Proxy settings in app. It's going out in next release which is tentatively v4.6 on Mac, scheduled for today. Would love if anyone can confirm this once the update is received.

**Comment by shamasis** [comment]
This should be resolved now. Closing. Please reopen if otherwise.

**Comment by DerFlash** [comment]
Works like a charm. Thanks a lot!

**Comment by everydaynerd** [comment]
@shamasis This proxy issue is still happening for myself and others in the collection runner.  The normal postman works great now behind a proxy (thanks for that) but the collection runner is still not working behind proxy (works fine without).  Thanks!

**Comment by SamvelRaja** [comment]
@everydaynerd  Thanks for pointing this out. we had fixed this in our internal build, will be available in the next release.

**Comment by everydaynerd** [comment]
Awesome - thanks!

**Comment by SamvelRaja** [comment]
@everydaynerd  This should be fixed in the latest `4.6.1` release. kindly confirm this and close the issue.

**Comment by everydaynerd** [comment]
@SamvelRaja Outstanding!  That was FAST! :)  Well done, it's all working on our end. 

I don't see an option to close though...

**Comment by DerFlash** [comment]
You didn't, did you? ;-)

Just downloaded the 4.7 update automatically and proxy support is completely broken again :-(

**Comment by SamvelRaja** [comment]
@DerFlash Can you get me a screen shot of our postmanConsole (View -> Show postman console or Cmd+Alt+C)
It will show the proxy details there.

**Comment by DerFlash** [comment]
Hey SamvelRaja,

here you go:
<img width="933" alt="bildschirmfoto 2016-09-06 um 08 27 25" src="https://cloud.githubusercontent.com/assets/185013/18263638/f13d48bc-740b-11e6-8251-779099153dbc.png">

It seems to read the proxy settings but I'd guess this little space in the proxy URL may be the issue here? I'd need to switch back to the 4.6 version to check if it's there too.

**Comment by SamvelRaja** [comment]
@DerFlash Kindly add the protocol in your proxy url. Will fix the problem now. We will update this in our next release.
`http://10.100.253.28` instead of `10.100.253.28`

**Comment by DerFlash** [comment]
@SamvelRaja Tried that. Hope I did it right:
<img width="609" alt="bildschirmfoto 2016-09-06 um 08 39 02" src="https://cloud.githubusercontent.com/assets/185013/18263967/a61e7534-740d-11e6-98b1-f19418624582.png">

Unfortunately no website can be opened afterwards. The proxy URL in postman though looks correct, but it throws some other error:
<img width="945" alt="bildschirmfoto 2016-09-06 um 08 37 50" src="https://cloud.githubusercontent.com/assets/185013/18263993/c771af30-740d-11e6-817a-0c6f2864e8f5.png">

Maybe this is because I'm using a VPN here?

~~Is there any chance to download the v4.6 again, so I could wait for the fix and work with the old version until then? I'm really depending on Postman at work here _g_~~
Nevermind, just edited the current download link ;-)

**Comment by SamvelRaja** [comment]
@DerFlash  can you mail us help@getpostman.com to sort out the things.

**Comment by DerFlash** [comment]
All good. 4.6.3 works like a charm again here. I'm just waiting for the update then.

Thanks a lot for your quick response to the ticket and your work for sure!

**Comment by pnt** [comment]
Broken again in 4.7.0. The URL (not visible in screenshot) is a publicly visible address, which requires me to go through the proxy. It's not 127.0.0.1 or internal.
![screen shot 2016-09-13 at 11 07 39](https://cloud.githubusercontent.com/assets/1242945/18466400/3a61c962-79a4-11e6-8b32-0a5ddbd59dcd.png)

Edit: Fixed in 4.7.1 which just arrived.

**Comment by everydaynerd** [comment]
I can confirm that 4.7.1 fixed the proxy issues we were seeing as well.  Thanks!

**Comment by SamvelRaja** [comment]
@pnt @DerFlash The fix is there in our 4.7.1 version of the app kindly check the same from your end 😄

**Comment by esseti** [comment]
does it work with socket proxy?

**Comment by shamasis** [comment]
What's a secret proxy? You mean a proxy not defined or announcing itself in OS or you mean http://secretproxy.org/

**Comment by esseti** [comment]
socks https://www.dropbox.com/s/h6p79huyrhdv5ic/Screenshot%202016-10-04%2020.19.20.png?dl=0

**Comment by knvpk** [comment]
Same problem for me too, 
version 4.8.1
linux / x86-64
Chrome 54.0.2840.90

We have an office internal network with all requests proxied through single server. when we are trying to sync its not working. in out ubuntu system we have set all the environment variables required for proxy still not connecting.

**Comment by sdnts** [comment]
@pavankumarkatakam Sync uses Websockets. Does your internal proxy allow Websockets?

**Comment by knvpk** [comment]
Well I want to check with my administrator, Thanks for reply @madebysid

**Comment by viniciusffj** [comment]
I'm having a problem with proxy in version 4.9.2.

<img width="588" alt="screen shot 2016-12-16 at 1 58 27 pm" src="https://cloud.githubusercontent.com/assets/2095993/21274777/dcd5a2b2-c397-11e6-9acf-28d730e8f14d.png">

**Comment by numaanashraf** [comment]
@viniciusffj Thanks for reporting this. We will roll out a fix for this in the coming releases.

**Comment by viniciusffj** [comment]
@numaanashraf Ok, I'll wait for that, thanks

**Comment by dmumladze** [comment]
v4.9.3 has the same issue.

**Comment by Vishal-Joshi** [comment]
Whenever I set proxy in Google Chrome, any requests I make from Postman app hangs forever.

I am using Postman Version 4.9.3 (4.9.3) on Mac OS v10.11.6

@numaanashraf Any update on the fix? Meanwhile do you have a workaround?

**Comment by ataraxus** [comment]
Proxy settings are not considered with postman: 

```
Postman for Chrome
Version 4.10.2
OS X / x86-64
Chrome 56.0.2924.87
```

**Comment by magicdawn** [comment]
Having this problem too.
Native App 4.10.3 (4.10.3).

![image](https://cloud.githubusercontent.com/assets/4067115/24188581/e464ae9e-0f1c-11e7-99bf-c98b125b4a6e.png)

![image](https://cloud.githubusercontent.com/assets/4067115/24188603/f7529b74-0f1c-11e7-92c8-5338ee6655ff.png)


But chrome app works fine under this network config

**Comment by BrunoBerisso** [comment]
I also have the same problem. Native app 4.10.3, proxy settings doesn't work.

**Comment by ghost** [comment]
Same problem, Native app 4.11.1 for Mac, proxy settings won't work

**Comment by jeamland** [comment]
Confirmed in Postman 5.0.0 on macOS Sierra 10.12.5. I have a SOCKS proxy configured in the system proxy settings in Network. Let's call it `proxy.mycompany.com` and port 1080.

What I see in the Postman logs is:

```Error: getaddrinfo ENOTFOUND socks5proxy.mycompany.com socks5proxy.mycompany.com:1080```

and the proxy information given is:

```
Proxy:
   proxy: "http://SOCKS5proxy.mycompany.com:1080"
   tunnel: false
```

Which is pretty much exactly what @magicdawn reported a few posts ago.

It very much looks like Postman's not coping with that particular proxy mechanism.

**Comment by msanecki** [comment]
I have also similar problem with Postman standalone
FD607AB7BBF6F542681055A915787B6A24EADD54 Postman-5.1.2-full.nupkg 68928972
OS: Windows

Working behind system firewall - not possible: "Could not get any response"
Working without system proxy - no problems!

**Comment by Mr-ZeroX** [comment]
I use the native app and get the message `Could not get any response` when I use the proxy `Shadowsocks`.
I changed the proxies in the Network Preferences and it works now.

![wx20170727-112721](https://user-images.githubusercontent.com/3412003/28653271-3eece312-72bf-11e7-87e0-88a774e84852.png)

**Comment by msanecki** [comment]
It is really strange because the request goes trough the proxy occasionally. 
This is what I get in Postman console when it does not work:

GET
https://echo.getpostman.com/digest-auth
12:30:30.036
Error: tunneling socket could not be established, statusCode=407
Proxy:
host:"10.252.76.110"
port:8080
match:"http+https://*/*"
tunnel:true
Request Headers:
cache-control:"no-cache"


and by the second call it works fine:

GET
https://echo.getpostman.com/digest-auth
12:33:08.215
Proxy:
host:"10.252.76.110"
port:8080
match:"http+https://*/*"
tunnel:true
Request Headers:
cache-control:"no-cache"
User-Agent:"PostmanRuntime/6.2.5"
Accept:"*/*"
accept-encoding:"gzip, deflate"
referer:"https://echo.getpostman.com/digest-auth"
...
Is it a bug?

**Comment by harryi3t** [comment]
In the latest version of Postman 5.2.0, we have added additional settings to manage proxy settings
![image](https://user-images.githubusercontent.com/5207331/29882136-c65ccb12-8dc9-11e7-99de-516cdbe2529e.png)

Now user can either use the `System Proxy` (defined at the OS level) or define a new proxy just for Postman to use.
Here's the link for the documentation about using proxy settings
https://go.pstmn.io/proxy-doc



Let me know if you still face the issue.

**Comment by devblaz** [comment]
I am behind a company proxy which requires authentication.
Could you please add functionality to authenticate on proxy servers?

**Comment by sdnts** [comment]
@devblaz Can you try using the System proxy the way @harryi3t has explained and check if things work?

**Comment by devblaz** [comment]
Hi @madebysid,
I have. I am using Postman 5.3.0 and have enabled "Use System Proxy".
When i try to reach a webservice on the internet, i get 407 authorization error from our company proxy.

We will be working around this problem so this is not blocking me at the moment. But i think it would be a great improvement of Postman if the already improved proxy-support would be further enhanced with authentication.
Thanks.

**Comment by harryi3t** [comment]
@devblaz Yes, that is expected. The operating system does not expose us the proxy credentials.
This would work when we have authentication for custom proxy (proxy defined in app), for which there is a feature request here https://github.com/postmanlabs/postman-app-support/issues/3566

**Comment by devblaz** [comment]
Ah that explains.
Great to hear there is already a feature request for that. I could not find it, sorry i must have overlooked it.
Thanks for your answer.

**Comment by abionescu** [comment]
**Postman for Windows
Version 5.3.2
win32 10.0.15063 / x64**

> connection using manually defined proxy (for requests) working as expected.

Proxy:
host:"127.0.0.1"
port:8888
match:"http+https://*/*"
Request Headers:
cache-control:"no-cache"
postman-token:"ff0f6276-e3ed-403f-8263-70fe3a7088bb"
user-agent:"PostmanRuntime/6.4.1"
accept:"*/*"
host:"postman-echo.com"
cookie:"sails.sid=s%3ArUJcR0l8iPAxmjV5vO_TD5LwIiM4fKOI.y3eQ2mVrxJrmO6g5LumhExdBjJ%2BrDZ7OxHRQ4fc72b4"
accept-encoding:"gzip, deflate"
Response Headers:
access-control-allow-credentials:""
access-control-allow-headers:""
access-control-allow-methods:""
access-control-allow-origin:""
access-control-expose-headers:""
content-encoding:"gzip"
content-type:"text/html; charset=utf-8"
date:"Wed, 08 Nov 2017 10:43:21 GMT"
etag:"W/"1d-3wD8DP6oX9PguFACSKPe4w""
server:"nginx"
vary:"Accept-Encoding"
content-length:"49"
connection:"Close"

> Collection  sync is **NOT** working for standalone app

- in Chrome status is `IN SYNC`
- in standalone app is flipping continuously between `CONNECTING` and `SYNCING`.

All the best,

**Comment by cdechery** [comment]
Proxy support is broken in 5.3.2.

It was working fine and then it stopped, and it won't work again no matter what I do. I tried every possible settings combination for this. Even uninstalled Postman and installed it again. Nothing.

I have disabled "Global Proxy" to make it use the System Proxy, but the console still shows its attempts at connecting to the IP configured in Global Proxy. The "System Proxy" setting is impossible to use.

![image](https://user-images.githubusercontent.com/5033136/33317554-9c463e94-d41e-11e7-91cb-ef596171c633.png)

![image](https://user-images.githubusercontent.com/5033136/33317573-aa475d34-d41e-11e7-9372-d4e63ff25760.png)

**Comment by chrisrymer** [comment]
Same issue: 407s over authenticated proxy

**Comment by calabresechris** [comment]
Same issue here. Proxy doesn't work in 5.3.2

**Comment by barneydmedia** [comment]
I'm having the same issue in v5.5 for OSX. Curl and browsers can reach my dev server through OpenVPN, but Postman cannot.

**Comment by j9feng** [comment]
Same issue here. Proxy doesn't work in 5.3.2

**Comment by bhanupinninti** [comment]
same issue for me in latest 6.3.2 version...running from postman runner is throwing 403 socket error.. a behind the proxy
![image](https://user-images.githubusercontent.com/38074414/41137447-868f3848-6b1e-11e8-8ba6-c2d55e6f7af1.png)
